### PR TITLE
Near-plugins integration

### PR DESCRIPTION
**Pull request by karim-en**
Integrate near plugins into eth-client, eth2-client, and prover contracts.

### CODE REVIEW

File: contracts/near/eth2-client/src/lib.rs [hunk 1]
```diff
@@ -603,8 +614,6 @@ impl Eth2Client {
     }
 
     fn is_light_client_update_allowed(&self) {
```
Comment (olga24912) [inline contracts/near/eth2-client/src/lib.rs:616]: `#[pause(except(owner, self))]` ?

**Review by olga24912 — CHANGES_REQUESTED** [review]

File: contracts/near/eth2-client/src/lib.rs (hunk 1 above)
Comment (karim-en) [reply contracts/near/eth2-client/src/lib.rs:616]: This is an internal method, so the pausing check is moved to the top-level method `submit_beacon_chain_light_client_update`

File: contracts/near/eth-prover/src/lib.rs [hunk 2]
```diff
@@ -142,6 +169,7 @@ impl EthProver {
     /// the StorageProof `value` field.  In order to verify the proof of non
     /// existence, you must set `value` to empty vec, *not* the RLP encoding of 0 or null
     /// (which would be 0x80).
+    #[pause(except(roles(Role::UnrestrictedVerifyLogEntry)))]
```
[the remark this answers is not in the document]
Comment (olga24912) [reply contracts/near/eth-prover/src/lib.rs:172]: add DAO role as well?

File: contracts/near/eth-prover/src/lib.rs [hunk 3]
```diff
@@ -96,6 +120,7 @@ impl EthProver {
             .into()
     }
 
+    #[pause(except(roles(Role::UnrestrictedVerifyLogEntry)))]
```
Comment (olga24912) [inline contracts/near/eth-prover/src/lib.rs:123]: add DAO role as well?

**Review by olga24912 — CHANGES_REQUESTED** [review]

File: contracts/near/eth-prover/src/lib.rs (hunk 3 above)
Comment (karim-en) [reply contracts/near/eth-prover/src/lib.rs]: d037e4e196c3232b8cbbd55b1574b75085b8310e

File: contracts/near/eth-prover/src/lib.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (karim-en) [reply contracts/near/eth-prover/src/lib.rs]: d037e4e196c3232b8cbbd55b1574b75085b8310e

File: contracts/near/eth-client/src/lib.rs [hunk 4]
```diff
@@ -198,14 +219,14 @@ impl EthClient {
     /// Add the block header to the client.
     /// `block_header` -- RLP-encoded Ethereum header;
     /// `dag_nodes` -- dag nodes with their merkle proofs.
+    #[pause(except(roles(Role::UnrestrictedAddBlockHeader)))]
```
[the remark this answers is not in the document]
Comment (karim-en) [reply contracts/near/eth-client/src/lib.rs]: d037e4e196c3232b8cbbd55b1574b75085b8310e

**Review by olga24912 — APPROVED** [review]

**Pull request merged**